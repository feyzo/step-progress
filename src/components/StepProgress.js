import React from 'react';
import styled from 'styled-components';
import Step from './Step';

const Container = styled.div`
  flex: 1;
  display: flex;
  flex-direction: row;
`;

function StepProgress(props) {
  const { steps, onStepChange } = props;
  const stepCount = steps.length;

  return (
    <Container>{
      steps.map((s, i) => {
        const next = steps[i + 1];

        return (
          <Step
            name={ s.name }
            complete={ s.complete }
            showLine={ i !== stepCount - 1 }
            completeLine={ next && next.complete }
            key={ s.name }
            onClick={onStepChange.bind(null, s)}
          />)
        })
    }</Container>
  );
}

export default StepProgress;

