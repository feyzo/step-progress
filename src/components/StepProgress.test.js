import React from 'react';
import renderer from 'react-test-renderer';
import StepProgress from './StepProgress';

jest.mock('./Step', () => 'Step');

const steps = [{
  name: 'step 1',
  complete: true
}, {
  name: 'step 2',
  complete: true
}, {
  name: 'step 3',
  complete: false
}]

test('Renders three steps', () => {
  const component = renderer.create(
    <StepProgress
      steps={ steps }
      onStepChange={ jest.fn() }
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});