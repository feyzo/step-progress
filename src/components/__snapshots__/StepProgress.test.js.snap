exports[`test Renders three steps 1`] = `
<div
  className="vZaAr">
  <Step
    complete={true}
    completeLine={true}
    enabled={undefined}
    name="step 1"
    onClick={[Function]}
    showLine={true} />
  <Step
    complete={true}
    completeLine={false}
    enabled={undefined}
    name="step 2"
    onClick={[Function]}
    showLine={true} />
  <Step
    complete={false}
    completeLine={undefined}
    enabled={undefined}
    name="step 3"
    onClick={[Function]}
    showLine={false} />
</div>
`;
